kvdb/postgres/db_test.go

```diff
@@ -0,0 +1,199 @@
+package postgres
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+func TestDb(t *testing.T) {
```

Comment:
Would it be possible to break this test up a little bit to have more focus on individual functionalities of the `kvdb` interface? So testing the txn, buckets, sequences, cursors etc. Also I think it's worth actually testing that the DB contents are what we'd expect (if we weren't assuming the implementation to be a black box).

Reply:
This test is definitely not in its final form. This is an interface test, so I should probably merge things with `walletdbtest`.

>My main concern is that testing should be a bit more focused since we can't expect to test the DB implementation with the itests (although it's a pretty good measure too).

How would you test the db implementation if it isn't in an itest with an actual postgres instance? I doubt that testing with a mock db driver and catching queries is really giving us the coverage that we want.

Reply:
Or what would you think of making the current etcd tests for buckets, cursors, etc backend independent?

Reply:
@bhandras if you take a look at the commit "kvdb/test: add postgres test", you can see how the raw data assertions turned out.